### DISCUSSION TITLE
Unable to remove rustaceanvim.mason warning

### DISCUSSION THREAD

**Discussion by AnirudhG07** [mrcjkb/rustaceanvim#551] [Q&A]
### Checked other resources

- [X] This is not a bug report or feature request (Please use [issues](https://github.com/mrcjkb/rustaceanvim/issues/new/choose) for those).
- [X] This is a question about rustaceanvim, not rust-analyzer or Neovim.
- [X] This is [not a question about autocompletion](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#im-having-issues-with-autocompletion).
- [X] I have read the README.
- [X] I have read the [FAQ](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#left_speech_bubble-faq).
- [X] I have searched for a solution using `:h rustaceanvim`.
- [X] I have searched [existing discussions](https://github.com/mrcjkb/rustaceanvim/discussions).
- [X] I have searched the existing issues/discussions of other plugins related to this question (if applicable).
- [ ] I have searched the [existing rust-analyzer discussions](https://github.com/rust-lang/rust-analyzer/discussions) (if applicable).
- [X] I have added a very descriptive title to this question.

### Description

I am trying to setup my rust DAP. Before this I had configured `rust_analyser` in mason.lua.
It is showing constantly -
```
nvim-lspconfig.rust_analyzer has been setup.
This will likely lead to conflicts with the rustaceanvim LSP client.
See ':h rustaceanvim.mason'
```
As it said, I made this rustaceanvim.lua file to retain the configs in mason.lua and remove the warning - 

```lua
return {
	"mrcjkb/rustaceanvim",
	version = "^5", -- Recommended
	lazy = false, -- This plugin is already lazy

	config = function()
		vim.g.rustaceanvim = {
			server = {
				cmd = function()
					local mason_registry = require("mason-registry")
					local ra_binary = mason_registry.is_installed("rust-analyzer")
							-- This may need to be tweaked, depending on the operating system.
							and mason_registry.get_package("rust-analyzer"):get_install_path() .. "/rust-analyzer"
						or "rust-analyzer"
					return { ra_binary } -- You can add args to the list, such as '--log-file'
				end,
			},
		}
	end,
}
```
But it is still not dissapearing. 
I tried the below combinations- ( X means didnt work) 
1) Above - X
2) removed any rust_analyser mentioned from mason lspconfig and added ABOVE - X
3) removed any rust_analyser... and only use the README.md mentioned 3 line `rustaceanvim.lua` - X

I donno what to do, I reinstalled rust-analyser just in case. Still didnt work. Please help me setup LSP and DAP for Rust!

### Example Code

```lua
I am putting links to my [mason.lua](https://github.com/AnirudhG07/dotfiles/blob/ugcl/nvim/.config/nvim/lua/anirudh/plugins/lsp/mason.lua), [lspconfig.lua](https://github.com/AnirudhG07/dotfiles/blob/ugcl/nvim/.config/nvim/lua/anirudh/plugins/lsp/lspconfig.lua) and [rustaceanvim.lua](https://github.com/AnirudhG07/dotfiles/blob/ugcl/nvim/.config/nvim/lua/anirudh/plugins/rustaceanvim.lua) file. Please suggest changes based on them. I have no clue what to do.
It would be appreciated for some improvements to rustaceanvim.lua file!
```

### Neovim version (nvim -v)

v0.10.2

### Operating system/version

Ubunty 22.04.4

### Output of :checkhealth rustaceanvim

```console
==============================================================================
rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- OK [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap) installed.

Checking external dependencies ~
- ERROR Failed to run healthcheck for "rustaceanvim" plugin. Exception:
  ...share/nvim/lazy/rustaceanvim/lua/rustaceanvim/health.lua:96:       rust-analyzer: not found: Could not find an executable binary.
        rustaceanvim requires [rust-analyzer](https://rust-analyzer.github.io/).
        Required by the LSP client.
```

**Comment by mrcjkb** [comment]
Duplicate of https://github.com/mrcjkb/rustaceanvim/discussions/94
